Classes/IndexQueue/Queue.php

```diff
@@ -351,7 +351,15 @@ private function addNewItem($itemType, $itemUid, $indexingConfiguration, $rootPa
             $additionalRecordFields = ', doktype, uid';
         }
 
-        $record = BackendUtility::getRecord($itemType, $itemUid, 'pid' . $additionalRecordFields);
+        $cache = GeneralUtility::makeInstance(TwoLevelCache::class, 'cache_runtime');
```

Comment:
I would propose to move the code into a protected method "getRecordCached" not enhance the lengh of the original method.